### PR TITLE
Add test directories to gitignore

### PR DESCRIPTION
**Pull request by mnonnenmacher**
Add directories which are created during running the functional tests to
gitignore.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -3,3 +3,5 @@
 build/
 *.iml
 /local.properties
+/plugin/out
```
Comment (sschuberth) [inline .gitignore]: Nit: To emphasize that we intend to match directories, we usually use a trailing slash and sort them in after the other directories at the beginning of this file, i.e. after `/.idea/`.

Comment (mnonnenmacher) [reply .gitignore]: Done.

File: .gitignore [hunk 2]
```diff
@@ -1,5 +1,8 @@
 /.gradle/
 /.idea/
+/plugin/out/
+/target/
+
```
Comment (sschuberth) [inline .gitignore]: Is the blank line here now intended? Because `build/` below is also a directory, it's just not limited to match in the root.

Comment (mnonnenmacher) [reply .gitignore]: Removed the blank line.

**Pull request merged**